### DISCUSSION TITLE
Feature request: Query time shift / specify time interval

### DISCUSSION THREAD

**Discussion by p4block** [grafana/grafana-csv-datasource#176] [Ideas]
In our old data visualization system (a bunch of JS and HighCharts) we split the time series into multiple ones by their year, so we could plot a year-over-year chart

![image](https://user-images.githubusercontent.com/5960575/111986742-a0e38380-8b0e-11eb-88be-986e365749c4.png)
Like this

I'm thinking how I could replicate a similar thing in Grafana and the built-in timeshift feature was implemented by the datasource. I would probably try stacking multiple -1y, -2y, -3y etc queries. It would be manual, but it would work.

Outputting multiple time series directly by a specified time interval would be best but I don't know if that would be outside scope of this plugin. 

(The new Grafana time series transformations could have something like this but as far I can see, they don't in their current featureset. In fact, they seem to be made with the opposite idea: merging series. )

**Comment by marcusolsson** [comment]
Splitting query results into multiple series is something that's been requested for my other data source as well. This will definitely happen as soon as I figure out the UX for it. Ideally I'd like for the data sources to work for any data—not only time series. 

When it comes to time shifts, I believe this is a generic feature to all data sources which means there's a chance we'll see improved support in Grafana itself. While I recognize that it would be a useful feature, I probably will be prioritizing feature that are more specific to this plugin. I might reconsider if this turns out to be frequented requested.
